lib/src/http.rs

```diff
@@ -221,9 +221,15 @@ impl HttpSession {
     fn upgrade_http(&mut self, http: Http<TcpStream, HttpListener>) -> Option<HttpStateMachine> {
         debug!("http switching to ws");
         let front_token = self.frontend_token;
-        let back_token = unwrap_msg!(http.backend_token);
-        let ws_context = http.websocket_context();
+        let back_token = match http.backend_token {
```

Comment:
I like it!
Please remove line 48 of the same file: ` util::UnwrapLog,`, since it is unused now